### ISSUE TITLE
Support for Multiple Headers with the Same Name

### ISSUE THREAD

**Issue by ganesh-bruno** [usebruno/bruno#3475]
### I have checked the following:

- [X] I've searched existing issues and found nothing related to my issue.

### Describe the feature you want to add

Currently, Bruno does not support sending multiple headers with the same name but different values. 

In the Bruno app, when sending multiple headers with the same name but different values, only the last header is shown in the timeline. The previous headers with the same name are not displayed, making it difficult to verify all sent headers.

### Mockups or Images of the feature

Here's the screenshot. 

<img width="896" alt="multiple-headers" src="https://github.com/user-attachments/assets/3d7b1c5e-6ca3-4447-8036-1bf34cc3325e">

**Comment by hosadoya** [comment]
Yes, this is a major issue. PostMan handles this correctly by not overwriting and sending multiple headers.

**Comment by volker-schukai** [comment]
I'm not sure if this is the same problem, but with set-cookie there are different headers. This is also described in the specification.

I'm having this problem. The server sends multiple `set-cookies` and in curl the headers are displayed correctly. In the bruno web interface, the values are somehow combined.

### curl

```
< content-type: application/json
< set-cookie: session=; Path=/; Max-Age=0
< set-cookie: session=secret; Path=/; Max-Age=3600
``` 

### bruno

Somehow it seems to be combined.

```
set-cookie	session=; Path=/; Max-Age=0session=secret; Path=/; Max-Age=3600
```

**Comment by supriyahegde** [comment]
Hi Team/ @sreelakshmi-bruno ,

We are also facing this issue. 
We receive multiple set-cookie headers in response header with different values and both are needed to be sent in further requests.
Could you please support this feature in Bruno.

Thanks,
Supriya

**Comment by justinTangren** [comment]
I ran into this today, My application supports a custom `X-FS-Feature-Tag` header. We support sending this header either as a comma seperated list, and also as multiple headers each with a diffenrt value

Assuming that bruno had sent both `X-FS-Feature-Tag` headers I had defined, I wasted several minutes hunting for a non-exising bug in my code before I realized that only one of the `X-FS-Feature-Tag` headers had been sent.

Could you please support this feature in Bruno.

**Comment by benharri** [comment]
> ... unless the entire field value for that header field is defined as a comma-separated list [i.e., #(values)]

Specifically for the Accept header, it's defined as #(values) so these should be parsed identically:

```
Accept: text/plain, application/json
```

```
Accept: text/plain
Accept: application/json
```

**Comment by helloanoop** [comment]
Although the RFC specifies that multiple headers should not be sent, some systems may not strictly follow this standard. Given Bruno’s large user base, we plan to explore options to support such cases. We don’t want to disadvantage users just because the server implementation isn’t fully compliant with the RFC.

We also need to see the impact to the header api's
* `req.getHeader()`, `req.getHeaders()`
* `req.setHeader()`, `req.setHeaders()`
* `res.getHeader()`, `res.getHeaders()` 

Quick search suggests that axios does not support sending duplicate headers, that makes the effort more harder.
I have added this to our work backlog for further refinement on options. It'll be some time before we get to this. If anyone in the community wants to take a stab at this, please do.

**Comment by justinTangren** [comment]
> Although the RFC specifies that multiple headers should not be sent, some systems may not strictly follow this standard

The RFC Specification says does NOT say multiple headers with the same name should never be sent. It does qualify when doing so is acceptable. The **unless** claus clearly outlines that sending multiple headers with the same name is allowed if the header is defined as a comma-separated list. 

From https://www.rfc-editor.org/rfc/rfc7230#section-3.2.2

> A sender MUST NOT generate multiple header fields with the same field
> name in a message unless either the entire field value for that
> header field is defined as a comma-separated list [i.e., #(values)]
> or the header field is a well-known exception (as noted below).
> 
> A recipient MAY combine multiple header fields with the same field
> name into one "field-name: field-value" pair, without changing the
> semantics of the message, by appending each subsequent field value to
> the combined field value in order, separated by a comma. . . 



---

> We also need to see the impact to the header api's

when developing in Java the javax.servlet.http.HttpServletRequest class behaves as follows when there are multiple headers with the same name
- Method [getHeader(String name)](https://jakarta.ee/specifications/servlet/5.0/apidocs/jakarta/servlet/http/httpservletrequest#getHeader(java.lang.String))
> Returns the value of the specified request header as a String. . . If there are multiple headers with the same name, this method returns the first head in the request

- method [getHeaders​(String name)](https://jakarta.ee/specifications/servlet/5.0/apidocs/jakarta/servlet/http/httpservletrequest#getHeaders(java.lang.String))
> Returns all the values of the specified request header as an Enumeration of String objects.

The bruno header api's could do something similar.
- Implement one method that returns only a first header value
- Implement another method that returns all values of the header

another appoach I have seen used is to have the `getHeader(String name)` method automatically concatenate the values into a comma seperated list if there are multiple headers with the same name

**Comment by justinTangren** [comment]
> another appoach I have seen used is to have the getHeader(String name) method automatically concatenate the values into a comma seperated list if there are multiple headers with the same name

While I have seen some of the spring boot annotations automatically combine multiple headers into a single comma-sepperated list, I think that having a getHeaders(String name) method that returns all the values as an enumeration would be better since the `Set-Cookie` header seems like it may break if combined into a single value

from [the same RFC link](https://www.rfc-editor.org/rfc/rfc7230#section-3.2.2) that was quoted earlier in this thread
> Note: In practice, the "Set-Cookie" header field ([[RFC6265](https://www.rfc-editor.org/rfc/rfc6265)]) often
      appears multiple times in a response message and does not use the
      list syntax, violating the above requirements on multiple header
      fields with the same name.  Since it cannot be combined into a
      single field-value, recipients ought to handle "Set-Cookie" as a
      special case while processing header fields.

This stack overflow comment (last edited in 2021) also contains many links explaining why multiple `Set-Cookie` header fields should not be folded into a single header field  
https://stackoverflow.com/a/6375214

**Comment by imyrke** [comment]
I noted that in the current release we are now able to have multiple keys with the same name in the body (url encoded). This was also not possible before and is a very useful change for us. The current is about headers and I wonder if that might have been fixes as well?

**Comment by SpaceFox** [comment]
Hi there,

I had the case today: a user of my API does not follow the RFC which triggered a bug; and as I cannot change the incoming HTTP requests, I have to fix the bug and handle these malformed requests. Without the capability to sent multiple HTTP headers with the same name, I cannot test my fix using Bruno (v3.5.1 and v4.0.0).

In addition, **nothing in the UI warns about the limitation**, there are multiple headers in the table:

<img width="426" height="205" alt="Image" src="https://github.com/user-attachments/assets/a6c2a57d-817a-4e8b-a896-04317839b22b" />

But I had to go to the Timeline view to discover that actually only one of the `X-Forwarded-Host` has been sent:

<img width="708" height="403" alt="Image" src="https://github.com/user-attachments/assets/edc8f73c-7e61-4a37-92dc-54690a59a833" />
